src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -916,10 +917,35 @@ private void OutputListToOutputWindow(string prefix, IEnumerable<string> values)
                     continue;
                 }
 
-                if (addedReferencePaths.Add(path))
+                if (metadataReferencesToAdd.TryGetValue(path, out var existingProperties))
                 {
-                    projectContext.AddMetadataReference(path, reference.Properties);
+                    // Merge existing aliases the properties that are already there
+                    var allAliases = existingProperties.Aliases.AddRange(reference.Properties.Aliases);
+
+                    // If one is empty and the other isn't, then we need to toss the global alias in too. The other cases
```

Comment:
The compiler API allows you to have a metadata reference with 'global' in the aliases list, or an empty aliases list, which is implied to mean 'global'. This is meaning we handle the latter as the former since we could be getting either kind from the project system.